sqlite.go

```diff
@@ -130,6 +131,18 @@ func OpenConn(path string, flags OpenFlags) (*Conn, error) {
 	return openConn(path, flags)
 }
 
+func Crypto(conn *Conn, key string) error {
+	ckey := C.CString(key)
+	defer C.free(unsafe.Pointer(ckey))
+
+	// convert CString to void*: https://jamesadam.me/2016/03/26/c-and-go-dealing-with-void-parameters-in-cgo/
+	res := C.sqlite3_key(conn.conn, unsafe.Pointer(&ckey), 0)
```

Comment:
There are two things wrong with this call that were likely causing you some issues. First, the function signature in C is this
```c
int sqlite3_key(
   sqlite3 *db,        /* The connection from sqlite3_open() */
   const void *pKey,   /* The key */
   int nKey            /* Number of bytes in the key */
);
```
First note that by passing `ckey` is already a pointer. So by passing `&ckey` you are passing a pointer to a pointer. This is not something C can prevent you from. It has no idea you passed it the wrong thing, but this will produce unexpected behavior. 

The next parameter needs to be the length of the string or byte slice. By passing zero you are using an empty key.

Reply:
Thanks for pointing this out, I don't know what I was thinking. I have cleaned that up and also implemented Rekey. I still don't know the names. I came up with `ApplyKey` and `ApplyRekey`. I would gladly change it if you have a better names.